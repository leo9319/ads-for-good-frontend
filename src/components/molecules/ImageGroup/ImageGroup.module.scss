.imageGroupContainer {
  width: 100%;
  float: left;
  display: flex;
  gap: 24px;
  align-items: center;
  justify-content: center;
  flex-wrap: wrap;
}
