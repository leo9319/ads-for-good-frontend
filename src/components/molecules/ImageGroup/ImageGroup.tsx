import React, { ReactElement, useId } from 'react';
import { Image, ImageProps } from '@components/atoms/Image';
import styles from './ImageGroup.module.scss';
import { classNames, getModuleClasses } from '@utils/common/classNames';

export interface ImageGroupProps {
  /**
   * List of image to display in the group
   * @default []
   * @property
   */
  data: Array<ImageProps>;
  /**
   * Class name for image group container
   * @property
   */
  className?: string;
}

export const ImageGroup = ({
  data = [],
  className,
}: ImageGroupProps): ReactElement => {
  const key = useId();
  const classes = getModuleClasses(className?.trim(), styles);
  return (
    <div className={classNames(styles.imageGroupContainer, classes)}>
      {data?.map((props, index) => (
        <Image key={key + '-' + index} {...props} />
      ))}
    </div>
  );
};

export default ImageGroup;
