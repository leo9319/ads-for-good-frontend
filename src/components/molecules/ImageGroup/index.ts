export { default } from './ImageGroup';
export { ImageGroup } from './ImageGroup';
