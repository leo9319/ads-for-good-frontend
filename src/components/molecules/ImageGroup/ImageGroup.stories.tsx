import React from 'react';
import ImageGroup from './ImageGroup';
import { Meta, StoryObj } from '@storybook/react/*';
import AccreditedMember from '@assets/icons/logo/AccreditedMember.svg';
import BBB from '@assets/icons/logo/bbb.svg';
import ImagineCanada from '@assets/icons/logo/ImagineCanada.svg';
import hc from '@assets/icons/logo/hc.svg';
import FallbackImage from '@assets/images/FallbackImage.svg';

export const imageGroupStoriesData = [
  {
    src: AccreditedMember,
    alt: 'p1',
    fallbackSrc: FallbackImage,
    height: '40px',
    width: '126px',
    link: {
      href: 'https://www.google.com/',
      target: '_blank',
    },
  },
  {
    src: BBB,
    alt: 'p2',
    fallbackSrc: FallbackImage,
    height: '40px',
    width: '102px',
    link: {
      href: 'https://www.google.com/',
      target: '_blank',
    },
  },
  {
    src: ImagineCanada,
    alt: 'p3',
    fallbackSrc: FallbackImage,
    height: '40px',
    width: '40px',
    link: {
      href: 'https://www.google.com/',
      target: '_blank',
    },
  },
  {
    src: hc,
    alt: 'p4',
    fallbackSrc: FallbackImage,
    height: '40px',
    width: '135px',
    link: {
      href: 'https://www.google.com/',
      target: '_blank',
    },
  },
];

const meta = {
  title: 'Library/Molecules/ImageGroup',
  component: ImageGroup,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    data: {
      description: 'Image Group Data (Array of Image Objects)',
      control: {
        type: 'object',
      },
    },
    className: {
      control: 'text',
    },
  },
  decorators: [
    Story => (
      <div style={{ padding: '20px' }}>
        <Story />
      </div>
    ),
  ],
  excludeStories: ['imageGroupStoriesData'],
} satisfies Meta<typeof ImageGroup>;

export default meta;

type Story = StoryObj<typeof meta>;

export const partnerIcon: Story = {
  args: {
    data: imageGroupStoriesData,
  },
};
