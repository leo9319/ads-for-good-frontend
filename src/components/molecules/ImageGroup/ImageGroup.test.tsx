import React from 'react';
import ImageGroup from './ImageGroup';
import { render } from '@testing-library/react';
import { imageGroupStoriesData } from './ImageGroup.stories';

describe('ImageGroup', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<ImageGroup data={imageGroupStoriesData} />);
    expect(baseElement).toMatchSnapshot();
  });
});
