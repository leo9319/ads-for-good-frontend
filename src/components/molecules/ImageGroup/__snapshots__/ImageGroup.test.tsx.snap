// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGroup should render successfully 1`] = `
<body>
  <div>
    <div
      class="imageGroupContainer"
    >
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.google.com/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p1"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper"
          inert=""
          src="SvgrURL"
          style="--imgWidth: 126px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.google.com/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p2"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper"
          inert=""
          src="SvgrURL"
          style="--imgWidth: 102px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.google.com/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p3"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper"
          inert=""
          src="SvgrURL"
          style="--imgWidth: 40px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
      <a
        class="rt-Text rt-reset rt-Link rt-underline-auto link body-md"
        data-accent-color=""
        href="https://www.google.com/"
        tabindex="0"
        target="_blank"
      >
        <img
          alt="p4"
          aria-hidden="true"
          class="rt-Skeleton imgWrapper"
          inert=""
          src="SvgrURL"
          style="--imgWidth: 135px; --imgHeight: 40px;"
          tabindex="-1"
        />
      </a>
    </div>
  </div>
</body>
`;
